contracts/token/ERC20/ERC20.sol

```diff
@@ -229,74 +229,73 @@ contract ERC20 is Context, IERC20, IERC20Metadata {
     ) internal virtual {
         require(from != address(0), "ERC20: transfer from the zero address");
         require(to != address(0), "ERC20: transfer to the zero address");
+        _update(from, to, amount);
+    }
 
-        _beforeTokenTransfer(from, to, amount);
-
+    /**
+     * @dev Moves `amount` of tokens from `from` to `to`.
+     *
+     * Will mint (or burn) if `from` (or `to`) is the zero address.
+     *
+     * Emits a {Transfer} event.
+     *
+     */
+    function _update(
+        address from,
+        address to,
+        uint256 amount
+    ) internal virtual {
         uint256 fromBalance = _balances[from];
-        require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
-        unchecked {
-            _balances[from] = fromBalance - amount;
-            // Overflow not possible: the sum of all balances is capped by totalSupply, and the sum is preserved by
-            // decrementing then incrementing.
-            _balances[to] += amount;
+        if (from == address(0)) {
+            _totalSupply += amount;
+            unchecked {
+                // Overflow not possible: balance + amount is at most totalSupply + amount, which is checked above.
+                _balances[to] += amount;
+            }
+        } else if (to == address(0)) {
+            require(fromBalance >= amount, "ERC20: burn amount exceeds balance");
+            _totalSupply -= amount;
+            unchecked {
+                // Overflow not possible: balance - amount is at most totalSupply - amount, which is checked above.
+                _balances[from] = fromBalance - amount;
+            }
+        } else {
+            require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
+            unchecked {
+                // Overflow not possible: fromBalance - amount is at most 0, which is checked above.
```

Comment:
Not needed because of the `require` immediately above it. Sorry for the back and forth on this one, I realized our guidelines are actually that if there is the necessary `require` statement right before the unchecked operation we don't need to include a comment.

```suggestion
```